river/ensemble/adaptive_random_forest.py

```diff
@@ -303,25 +290,10 @@ def _new_learning_node(self, initial_stats=None, parent=None):   # noqa
             return new_adaptive
 
     def new_instance(self):
-        return self.__class__(max_size=self.max_size,
-                              memory_estimate_period=self.memory_estimate_period,
-                              grace_period=self.grace_period,
-                              split_confidence=self.split_confidence,
-                              tie_threshold=self.tie_threshold,
-                              binary_split=self.binary_split,
-                              stop_mem_management=self.stop_mem_management,
-                              remove_poor_attrs=self.remove_poor_attrs,
-                              merit_preprune=self.merit_preprune,
-                              leaf_prediction=self.leaf_prediction,
-                              leaf_model=self.leaf_model,
-                              model_selector_decay=self.model_selector_decay,
-                              max_features=self.max_features,
-                              nominal_attributes=self.nominal_attributes,
-                              attr_obs=self.attr_obs,
-                              attr_obs_params=self.attr_obs_params,
-                              min_samples_split=self.min_samples_split,
-                              max_depth=self.max_depth,    # noqa
-                              seed=self._rng)
+        params = self._get_params()
+        # Use existing rng to enforce a different model
+        params.update(seed=self._rng)
```

Comment:
Is this the reason why you document that `seed` can be a `np.random.RandomState`? Another possibility could simply be:

```py
new = self.clone()
new._rng = self._rng
```

Reply:
It is important to make sure that models are different inside the ensemble, we had some issues in the past where we were (wrongly) using the same empty models which could lead to under-performance. Using `RandomState` makes things easier.

I changed the code as suggested in cb3e7f2835520439edcd5d1b3f3be9bee18cf278